pkg/controller/datavolume-controller.go

```diff
@@ -2599,3 +2623,288 @@ func GetRequiredSpace(filesystemOverhead float64, requestedSpace int64) int64 {
+		}
+		// Create the pod
+		if err := r.client.Create(context.TODO(), pod); err != nil {
+			if !k8serrors.IsAlreadyExists(err) {
+				return nil, err
+			}
+		}
+
+		r.log.V(3).Info("Size-detection pod created\n", "pod.Name", pod.Name, "pod.Namespace", pod.Namespace)
+	}
+
+	return pod, nil
+}
+
+// makeSizeDetectionPodSpec creates and returns the full size-detection pod spec
+func makeSizeDetectionPodSpec(sourcePvc *corev1.PersistentVolumeClaim, client client.Client, pullPolicy string) *corev1.Pod {
+	image := os.Getenv("IMPORTER_IMAGE")
+	if image == "" {
+		return nil
+	}
+
+	workloadNodePlacement, err := GetWorkloadNodePlacement(client)
+	if err != nil {
+		return nil
+	}
+	// Generate individual specs
+	container := makeSizeDetectionContainerSpec(image, pullPolicy)
+	volume := makeSizeDetectionVolumeSpec(sourcePvc.Name)
+	objectMeta := makeSizeDetectionObjectMeta(sourcePvc, true, false)
+	// Assemble the pod
+	pod := &corev1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: *objectMeta,
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				*container,
+			},
+			Volumes: []corev1.Volume{
+				*volume,
+			},
+			RestartPolicy:     corev1.RestartPolicyOnFailure,
+			NodeSelector:      workloadNodePlacement.NodeSelector,
+			Tolerations:       workloadNodePlacement.Tolerations,
+			Affinity:          workloadNodePlacement.Affinity,
+			PriorityClassName: getPriorityClass(sourcePvc),
+		},
+	}
+
+	return pod
+}
+
+// makeSizeDetectionObjectMeta creates and returns the object metadata for the size-detection pod
+func makeSizeDetectionObjectMeta(sourcePvc *corev1.PersistentVolumeClaim, blockOwnerDeletion, isController bool) *metav1.ObjectMeta {
+	return &metav1.ObjectMeta{
+		Name:      sizeDetectionPodName(sourcePvc),
+		Namespace: sourcePvc.Namespace,
+		Annotations: map[string]string{
+			AnnCreatedBy: "yes",
+		},
+		Labels: map[string]string{
+			common.CDILabelKey:        common.CDILabelValue,
+			common.CDIComponentLabel:  common.ImporterPodName,
+			common.PrometheusLabelKey: common.PrometheusLabelValue,
+		},
+		OwnerReferences: []metav1.OwnerReference{
+			{
+				APIVersion:         "v1",
+				Kind:               "PersistentVolumeClaim",
+				Name:               sourcePvc.Name,
+				UID:                sourcePvc.GetUID(),
+				BlockOwnerDeletion: &blockOwnerDeletion,
+				Controller:         &isController,
+			},
+		},
+	}
+}
+
+// makeSizeDetectionContainerSpec creates and returns the size-detection pod's Container spec
+func makeSizeDetectionContainerSpec(image, pullPolicy string) *corev1.Container {
+	return &corev1.Container{
+		Name:            "size-detection-volume",
+		Image:           image,
+		ImagePullPolicy: corev1.PullPolicy(pullPolicy),
+		Command:         []string{"/usr/bin/cdi-image-size-detection"},
+		Args:            []string{"-url", common.ImporterWritePath, "-path", corev1.TerminationMessagePathDefault},
```

Comment:
I would make -path -termination-message-path so there is no confusion about what this parameter does. At first I thought it was the path to the disk.img file, but that is the -url parameter (which might be better as path)

Reply:
Yeah, that makes sense.